app/currency/arhive_rate.py

```diff
@@ -0,0 +1,80 @@
+from json import JSONDecodeError
+
+from currency.utils import to_decimal
+from currency.models import Rate, Source
+from currency import consts
+
+import requests
+from datetime import datetime
+
+
+DATE = f"{datetime.now():%d.%m.%Y}".split('.')
+
+
+def parse_url(day=DATE[0], month=DATE[1], year=DATE[2]):
+    count = 0
+    date_parse = []
+
+    while count != 3:
+
+        day = day if len(date_parse) == 0 else int(date_parse[0]) - 1
+        month = month if len(date_parse) == 0 else int(date_parse[1])
+        year = year if len(date_parse) == 0 else int(date_parse[2])
+
+        url = f"https://api.privatbank.ua/p24api/exchange_rates?json&date={day}.{month}.{year}"
```

Comment:
https://requests.readthedocs.io/en/latest/user/quickstart/#passing-parameters-in-urls